Formula/git-tools.rb

```diff
@@ -0,0 +1,39 @@
+class GitTools < Formula
+  include Language::Python::Shebang
+
+  desc "Assorted git-related scripts and tools"
+  homepage "https://github.com/MestreLion/git-tools"
+  url "https://github.com/MestreLion/git-tools/archive/v2022.12.tar.gz"
+  sha256 "b01e3b8de268ee07854e6cc66d78db6ed6cbc3e71dfeded8456138881bdf97f1"
+  license "GPL-3.0-only"
+  head "https://github.com/MestreLion/git-tools.git", branch: "main"
+
+  depends_on "python@3.11"
+  uses_from_macos "bash"
+
+  def install
+    rewrite_shebang detected_python_shebang, "git-restore-mtime"
+    bin.install Dir["git-*"]
+    man1.install Dir["man1/*.1"]
+  end
+
+  test do
+    assert_equal "git-restore-mtime version #{version}", shell_output("#{bin}/git-restore-mtime --version").chomp
+    system "git", "init"
+    system "git", "config", "user.name", "BrewTestBot"
+    system "git", "config", "user.email", "BrewTestBot@example.com"
+    touch "foo"
+    system "git", "add", "foo"
+    system "git", "commit", "-m", "foo"
+    touch "foo"
+    output = shell_output("#{bin}/git-restore-mtime . 2>&1")
+    assert_match "1 files to be processed in work dir", output
+    assert_match "1 files updated", output
+    assert_match "master -> test", shell_output("#{bin}/git-branches-rename -n -v master test")
```

Comment:
This assertion is failing. Maybe it's being printed to stderr?

Reply:
Should be fine since the test is passing in macOS 13 and Linux. Unfortunately I don't have access to macOS 12, so I cannot test it. The specific script seems to use `git` and `awk`, so my best guess is that one of the two is responsible. I may try setting a dependancy to the brew version of `awk` to see if that works.

Reply:
It looks like the default branch on the test system was `main` and my test relied on it being `master` 😅. I've modified the test and it looks fine now.